02-peer-protocol.md

```diff
@@ -1006,12 +1191,26 @@ sign the resulting transaction (as defined in [BOLT #3](03-transactions.md)), an
    * [`signature`:`signature`]
    * [`u16`:`num_htlcs`]
    * [`num_htlcs*signature`:`htlc_signature`]
+   * [`commitment_signed_tlvs`:`tlvs`]
+
+1. `tlv_stream`: `commitment_signed_tlvs`
+2. types:
+    1. type: 0 (`splice_commitsigs`)
+    2. data:
+        * [`...*commitsigs`:`sigs`]
+
+1. subtype: `commitsigs`
+2. data:
+   * [`signature`:`commit_signature`]
+   * [`u16`:`num_htlcs`]
+   * [`num_htlcs*signature`:`htlc_signature`]
```

Comment:
We can't do that unfortunately, we won't have enough space in the `commit_sig` message since it's restricted to 65kB...

We can use several `commit_sig` messages though, one for each version of the commit tx with one tlv field to indicate which splice these signatures are valid for (probably identified with just the corresponding funding txid or using an implicit ordering by feerate). We can mandate that if N splices are ongoing, senders must send (N+1) `commit_sig` messages in a row and the receiver treats them together atomically.

We can do the same for `revoke_and_ack`, or use a single message since in that case there should be enough space left in the tlv stream.

Reply:
Happy with multiple messages, sure.

Reply:
Do we have an example of how splitting over multiple messages works elsewhere in the spec or would we have to design something specific for commit_sig to fit into 65kB chunks?